src/main/java/com/dalendev/finance/cryptobot/config/ApplicationBoot.java

```diff
@@ -17,33 +22,57 @@
 @Component
 public class ApplicationBoot implements ApplicationListener<ApplicationReadyEvent> {
 
-    private final ExchangeInfoAdapter adapter;
-    private final CryptoCurrencyService currencyService;
+    private Log logger = LogFactory.getLog(CryptoCurrencyService.class);
 
-    @Value("${maxExchangeSymbols:99999}")
-    private Integer maxExchangeSymbols;
+    private final ExchangeInfoAdapter exchangeAdapter;
+    private final CryptoCurrencyService currencyService;
+    private final AccountAdapter accountAdapter;
+    private final Integer maxExchangeSymbols;
 
     @Autowired
-    public ApplicationBoot(ExchangeInfoAdapter adapter, CryptoCurrencyService currencyService) {
-        this.adapter = adapter;
+    public ApplicationBoot(
+            ExchangeInfoAdapter exchangeAdapter,
+            CryptoCurrencyService currencyService,
+            AccountAdapter accountAdapter,
+            @Value("${maxExchangeSymbols:99999}") Integer maxExchangeSymbols) {
+        this.exchangeAdapter = exchangeAdapter;
         this.currencyService = currencyService;
+        this.accountAdapter = accountAdapter;
+        this.maxExchangeSymbols = maxExchangeSymbols;
     }
 
     @Override
     public void onApplicationEvent(final ApplicationReadyEvent applicationEvent) {
+        assertEnoughBNB();
+        importSymbols();
+    }
 
-        ExchangeInfo exchangeInfo = adapter.getExchangeInfo();
+    private void importSymbols() {
+        ExchangeInfo exchangeInfo = exchangeAdapter.getExchangeInfo();
 
         exchangeInfo.getSymbols().stream().parallel()
-            .map(Symbol::getSymbol)
-            .sorted()
-            .filter(symbol -> symbol.endsWith("BTC"))
-            .limit(maxExchangeSymbols)
-            .forEach(symbol -> adapter.importSymbolPriceHistory(symbol).stream()
-                    .map(price -> new PriceTicker(symbol, price))
-                    .forEach(currencyService::updateMarket)
-            );
+                .map(Symbol::getSymbol)
+                .sorted()
+                .filter(symbol -> symbol.endsWith("BTC"))
+                .limit(maxExchangeSymbols)
+                .forEach(symbol -> exchangeAdapter.importSymbolPriceHistory(symbol).stream()
+                        .map(price -> new PriceTicker(symbol, price))
+                        .forEach(currencyService::updateMarket)
+                );
 
         currencyService.setMarketReady();
     }
+
+    private void assertEnoughBNB() {
+        AccountInformation accountInformation = accountAdapter.getAccountInformation();
+        Double bnbBalance = accountInformation.getBalances().stream()
+                .filter(balance -> balance.getAsset().equalsIgnoreCase("BNB"))
+                .findFirst()
+                .orElseThrow(() -> new LowBalanceException("BNB balance 0, buy some before starting the bot"))
+                .getFree();
+
+        logger.warn(String.format("You current BNB balance is: %.8f, make sure is enough for trading",
```

Comment:
so if I have 1k BNB's I still get a warning?

Reply:
it's a bit difficult to understand how much BNB you need to pay for the fees. if you allow the bot to trade thousands of BTC per trade you'll probably need a lot of BNB